src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs

```diff
@@ -118,7 +118,15 @@ public AsyncTargetWrapper(Target wrappedTarget)
         /// <param name="overflowAction">The action to be taken when the queue overflows.</param>
         public AsyncTargetWrapper(Target wrappedTarget, int queueLimit, AsyncTargetWrapperOverflowAction overflowAction)
         {
-            RequestQueue = new AsyncRequestQueue(10000, AsyncTargetWrapperOverflowAction.Discard);
+#if NETSTANDARD2_0
+            // NetStandard20 includes many optimizations for ConcurrentQueue:
+            //  - See: https://blogs.msdn.microsoft.com/dotnet/2017/06/07/performance-improvements-in-net-core/
+            // Net40 ConcurrencyQueue can seem to leak, because it doesn't clear properly on dequeue
+            //  - See: https://blogs.msdn.microsoft.com/pfxteam/2012/05/08/concurrentqueuet-holding-on-to-a-few-dequeued-elements/
+            _requestQueue = new ConcurrentRequestQueue(10000, AsyncTargetWrapperOverflowAction.Discard);
+#else
+            _requestQueue = new AsyncRequestQueue(10000, AsyncTargetWrapperOverflowAction.Discard);
```

Comment:
another option would be to create a `CreateRequestQueue(bool lockingQeueue)`, and set it only in `InitializeTarget`, isn't?

The would prefer having the creation and "business" rules for the creation in one region, I think `InitializeTarget` would be the best place.